File: solutions/541.reverse-string-ii/reverse-string-ii.cpp
class Solution {
public:
    string reverseStr(string s, int k) {
        int i = 0, j, temp, n = s.size();
        while(i < n) {
            j = min(i + k - 1, n - 1);
            temp = j + k + 1;
            while(i < j) swap(s[i++], s[j--]);
            i = temp;
        }
        return s;
    }
};

File: solutions/541.reverse-string-ii/reverse-string-ii.java
class Solution {
    public String reverseStr(String s, int k) {
        StringBuilder res = new StringBuilder();
        char[] str = s.toCharArray();
        for(int i = 0; i < s.length(); i++) {
            if((i / k) % 2 == 0) res.insert(i - i % k, str[i]);
            else res.append(str[i]);
        }
        return res.toString();
    }
}